File: akatpp-x/akat_x_gpio_access.py
class Macro:
    def render(self, inv):
        ctx = akat.prepare(inv, required_args = ["object_name", "reg_name", "pin_name"])

        reg_name = ctx.reg_name + ctx.pin_name[0]
        pin_idx = ctx.pin_name[1:]

        return akat.transform(akat.render(self, object_name = ctx.object_name, pin_name = ctx.pin_name, reg_name = reg_name, pin_idx = pin_idx))

File: akatpp-x/akat_x_gpio_access.c
OBJECT$(${object_name}) {
    METHOD$(void set(u8 state), inline) {
        if (state) {
            ${reg_name} |= 1 << ${pin_idx};  // Set ${reg_name} of ${pin_name} to 1
        } else {
            ${reg_name} &= ~(1 << ${pin_idx});  // Set ${reg_name} of ${pin_name} to 0
        }
    }

    METHOD$(u8 is_set(), inline) {
        return ${reg_name} & (1 << ${pin_idx});  // Get value of ${reg_name} for ${pin_name}
    }
}
